test/rulesets/Base/array.jl

```diff
@@ -34,6 +34,21 @@ end
     @test dC ≈ view(H̄, :, 4:6)
 end
 
+@testset "reduce hcat" begin
+    A = randn(3, 2)
+    B = randn(3, 1)
+    C = randn(3, 3)
+    H, pullback = rrule(reduce, hcat, [A, B, C])
+    @test H == reduce(hcat, [A, B, C])
+    H̄ = randn(3, 6)
+    (ds, dh, (dA, dB, dC)) = pullback(H̄)
+    @test ds == NO_FIELDS
+    @test dh == NO_FIELDS
+    @test dA ≈ view(H̄, :, 1:2)
+    @test dB ≈ view(H̄, :, 3:3)
+    @test dC ≈ view(H̄, :, 4:6)
+end
```

Comment:
Are you able to use `ChainRulesTestUtils.rrule_test` here and below? It's safest to verify with finite differences in case the derivation is wrong.